clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java

```diff
@@ -162,6 +200,24 @@ public ListTopicsResult listTopics() {
     /**
      * List the topics available in the cluster.
      *
+     * <p>The following exceptions can be anticipated when calling {@code get()} on the futures obtained from the
+     * {@link ListTopicsResult#namesToListings() namesToListings()}, {@link ListTopicsResult#listings() listings()},
+     * {@link ListTopicsResult#names() names()} methods of the  returned {@code ListTopicsResult}</p>
+     *
+     * <ul>
+     *     <li>{@link org.apache.kafka.common.errors.UnknownTopicOrPartitionException}
+     *     if the authorized topic was in metadata first, but was not found during later look up</li>
+     *     <li>{@link org.apache.kafka.common.errors.LeaderNotAvailableException}
+     *     if leader is unavailable</li>
+     *     <li>{@link org.apache.kafka.common.errors.ReplicaNotAvailableException}
+     *     if the replica is not available</li>
+     *     <li>{@link org.apache.kafka.common.errors.TopicAuthorizationException}
+     *     if the authenticated user is not authorized to alter the topic</li>
```

Comment:
We don't use TopicAuthorizationException in listTopics

Reply:
I see in the source https://github.com/apache/kafka/blob/88307657dff0202804447d06eef6ecdc808590d8/core/src/main/scala/kafka/server/KafkaApis.scala#L996